### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by d-beschastnyi**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,51 @@
...
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list[Animal]) -> int:
+    result = 0
+    for animal in animals:
+        result += animal.feed()
+    return result
```
Comment (vsmutok) [inline app/main.py:51]: use `sum` and comprehension

**Review by vsmutok — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 2]
```diff
@@ -1 +1,48 @@
...
+        super().__init__(name, is_hungry)
+        self.appetite = appetite
+
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list[Animal]) -> int:
+    return sum(Animal.feed(animal) for animal in animals)
```
Comment (vasylhnatiuk) [inline app/main.py:48]:
```suggestion
    return sum(animal.feed() for animal in animals)
```

Comment (vasylhnatiuk) [reply app/main.py:48]: don't use parent class if it not needed

File: app/main.py [hunk 3]
```diff
@@ -1 +1,48 @@
...
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        return 0
+
+
+class Cat(Animal):
+    def __init__(self,
+                 name: str,
+                 appetite: int = 3,
```
Comment (vasylhnatiuk) [inline app/main.py:24]: image
`Cat `init should not receive `appetite ` as an argument

Comment (vasylhnatiuk) [reply app/main.py:24]: the same with dog

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

**Review by vasylhnatiuk — APPROVED** [review]

**Pull request closed without merging**